### ISSUE TITLE
(fixed) 2.9.0+ missing module.exports

### ISSUE THREAD

**Issue by danez** [evgenyrodionov/redux-logger#210]
Version 2.9 and up have a breaking change.
Version 2.8.2 had `module.exports` set, whereas 2.9+ doesn't. This is probably as `add-module-exports` is not adding `module.exports` when there are named exports.

This breaks this kind of code:

```js
if (__DEV__) {
    const logger = require('redux-logger')({
        level: 'info',
        collapsed: true,
        ...
    });

    middlewares.push(logger);
}
```

https://unpkg.com/redux-logger@2.9.0
https://unpkg.com/redux-logger@2.8.2

**Comment by JayAndCatchFire** [comment]
According to semver rules, this should have bumped up to version 3.0.0 since there were breaking changes to the API.

**Comment by evgenyrodionov** [comment]
Interesting.

We use https://github.com/59naga/babel-plugin-add-module-exports for this behavior, investigating now why it broken

**Comment by evgenyrodionov** [comment]
Fixed in [2.10.1](https://github.com/evgenyrodionov/redux-logger/releases/tag/2.10.1), thanks

**Comment by danez** [comment]
Thanks for the quick fix.

**Comment by danez** [comment]
Although I don't see anything has changed in 2.10.1, still no `model.exports`

**Comment by paibamboo** [comment]
+1 Issue still persists

**Comment by evgenyrodionov** [comment]
Fuck

**Comment by evgenyrodionov** [comment]
Fucking `prepublish`, fixed in 2.10.2. Sorry.

https://unpkg.com/redux-logger@2.10.2

**Comment by danez** [comment]
Thanks, works.

**Comment by evgenyrodionov** [comment]
By the way, does anyone knows how to fix that issue?

I use [babel-plugin-add-module-exports](https://github.com/59naga/babel-plugin-add-module-exports) and when redux-logger exported single function (as < 2.9.0), it works fine

In
```javascript
export default createLogger
```

Out
```javascript
exports.default = createLogger;
module.exports = exports['default'];
```

But now I exports default logger and default options
In 
```javascript
export {
  defaults,
  defaultLogger as logger,
};

export default createLogger;
```

Out
```javascript
exports.defaults = _defaults2.default;
exports.logger = defaultLogger;
exports.default = createLogger;
```

As you see, `module.exports = exports['default'];` has gone. Why?

cc @59naga

**Comment by pilaas** [comment]
@evgenyrodionov This behaviour is described in `babel-plugin-add-module-exports` README.

> This plugin follows the babel@5 behavior - add the module.exports if **only** the export default declaration exists

By the way, your fix for this issue breaks importing like `import { logger } from 'redux-logger` since `exports` with `logger` property is overwritten in line `module.exports = exports['default']`.

**Comment by mcrawshaw** [comment]
@evgenyrodionov 2.10.2 broke the new typings that were merged a few hours ago (was working with 2.10.1).  More than happy to update again once the export is locked down.

**Comment by LKay** [comment]
@evgenyrodionov How about just use [transform-es2015-modules-commonjs](https://babeljs.io/docs/plugins/transform-es2015-modules-commonjs/) and not force `module.exports = exports['default']` and leave ES6 module export as it actually is? Then just bump the version to 3.0.0 (it will be breaking change)

Currently node supports ES6 modules just fine.

**Comment by evgenyrodionov** [comment]
@LKay thanks for this plugin, I like it.
Problem not with nodejs, problem with this case (from first post):

```javascript
if (__DEV__) {
    const logger = require('redux-logger')({
        level: 'info',
        collapsed: true,
        ...
    });

    middlewares.push(logger);
}
```

If we drop support of `module.exports = exports['default']`, users need to write something like
```javascript
    const logger = require('redux-logger').default({
        level: 'info',
        collapsed: true,
        ...
    });
```

It's not good.

**Comment by evgenyrodionov** [comment]
Working on that issue to release fixed version ASAP.

Goal for the API in 3.0.0 (will be with breaking change)
```javascript
import logger from 'redux-logger'
import { createLogger } from 'redux-logger'
const logger = require('redux-logger') // BROKEN because I don't understand how to made it works
const { createLogger } = require('redux-logger')
```

So we need something like this
```javascript
export {
  defaults, // Object with default settings
  createLogger 
}

export default logger // logger with default settings
```

If anyone can help, I'll be very grateful

**Comment by LKay** [comment]
As long as it will be major version, breaking changes are acceptable. There is not much you can do with default export of ES6 module and node. The plugin you are using will work if you have only default export but will break your code if you also have another named exports.

ES6 modules and common js modules are not fully compatible by design so you have to pick one. You can hack it to force default import in node to work but I wouldn't hammer it too much.

**Comment by evgenyrodionov** [comment]
@LKay don't knew that, thanks. 

Ok, will release new version with named exports in 1-2 hours

**Comment by mcrawshaw** [comment]
I'll check back in tomorrow morning (about 10 hours away) to test the typings.

**Comment by evgenyrodionov** [comment]
@mcrawshaw I'll mention you when it's done. Ok?

**Comment by evgenyrodionov** [comment]
Released [3.0.0](https://github.com/evgenyrodionov/redux-logger/releases/tag/3.0.0).

cc @mcrawshaw @arusakov

**Comment by mcrawshaw** [comment]
PR pending for types:
https://github.com/DefinitelyTyped/DefinitelyTyped/pull/15346
